pkg/tracing/trace_handler.go

```diff
@@ -113,3 +115,61 @@ func (h *TraceHandler) ShowSpan(w http.ResponseWriter, req bunrouter.Request) er
 		"span": span,
 	})
 }
+
+//------------------------------------------------------------------------------
+
+func (h *TraceHandler) ShowTraceJSON(w http.ResponseWriter, req bunrouter.Request) error {
+	ctx := req.Context()
+
+	traceID, err := uuid.Parse(req.Param("trace_id"))
+	if err != nil {
+		return err
+	}
+
+	spans, err := SelectTraceSpans(ctx, h.App, traceID)
+	if err != nil {
+		return err
+	}
+
+	if len(spans) == 0 {
+		return httperror.NotFound("Trace %q not found. Try again later.", traceID)
+	}
+
+	tracepbSpans := make([]*tracepb.Span, len(spans))
+	for i, span := range spans {
+		tracepbSpans[i] = span.TracepbSpan()
+	}
+
+	resp := &tracepb.TracesData{
+		ResourceSpans: []*tracepb.ResourceSpans{{
+			// Here we should have resource attributes that are common for all spans.
+			// But in the database, we mix resource attributes and normal attributes
+			// together so the information about resource attributes is lost.
+			//
+			// Using nil here should work too.
+			Resource: nil,
+
+			ScopeSpans: []*tracepb.ScopeSpans{{
+				Scope: nil,
+				Spans: tracepbSpans,
+			}},
+
+			// InstrumentationLibrarySpans field is deprecated in favor of ScopeSpans.
+			// It will be removed in next versions.
+			InstrumentationLibrarySpans: []*tracepb.InstrumentationLibrarySpans{{
+				InstrumentationLibrary: nil,
+				Spans:                  tracepbSpans,
+			}},
+		}},
```

Comment:
@lmangani here is where we generate the response:
- OpenTelemetry deprecated `InstrumentationLibrarySpans` field and is replacing it with `ScopeSpans`. I populate both fields for now - they contain the same info.
- `Resource: nil` and `Scope: nil` are always `nil` since that information is merged into `span.Attrs` when attributes are stored in the db. Let me know if that is a problem and we can try to find some workarounds.